components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc

```diff
@@ -220,14 +221,16 @@ When a transaction that raises the `send_message_to_l1` syscall is included in a
 * Payload size
 * Payload (list of field elements)
 
-Consequently, the fee associated with a single L2→L1 message is:
+Consequently, the gas costs associated with a single L2→L1 message is:
 
 [stem]
 ++++
-\textit{gas_price}\cdot \text{calldata_cost}\cdot(3+\text{payload_size})
+\text{storage_write_cost} + \left(\text{da_calldata_cost} + \text{message_calldata_cost}\right)\cdot\left(3+\text{payload_size}\right)
 ++++
 
-where stem:[$\text{calldata_cost}$] is 512 gas per 32-byte word.
+Where stem:[$\text{da_calldata_cost}$] and stem:[$\text{message_calldata_cost}$] are 551 and 512 gas correspondingly.
+The reason messages appear twice in the fee formula (and pay both in stem:[$\text{da_calldata_cost}$] and stem:[$\text{message_calldata_cost}$]), 
```

Comment:
```suggestion
Messages appear twice in the fee formula, and pay both in stem:[$\text{da_calldata_cost}$] and stem:[$\text{message_calldata_cost}$]) because 
```